src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java

```diff
@@ -523,4 +524,214 @@ public void testApi_RemoveInspectionPort_InspectionPortDisappears() throws Excep
+    }
+
+    @Test
+    public void testApiUpdatePort() throws Exception {
+        //Arrange
+        final ArrayList<String> macAdresses = new ArrayList<String>(
+                Arrays.asList("ff:ff:ff:bb:aa:02", "ff:ff:ff:bb:aa:03"));
+
+        final ArrayList<String> ipAdresses = new ArrayList<String>(Arrays.asList("10.5.2.3", "10.5.3.4"));
+
+        final String newParentId = "NewParentId";
+
+        final String newDeviceOwnerId = "NewDeviceOwnerId";
+
+        this.redirApi = new SampleSdnRedirectionApi(this.txControl, this.em);
+
+        this.redirApi.registerPort(port);
+
+        PortEntity portEntity = this.redirApi.getPort(port.getElementId());
+        portEntity.setMacAddresses(macAdresses);
+        portEntity.setPortIPs(ipAdresses);
+        portEntity.setParentId(newParentId);
+        portEntity.setDeviceOwnerId(newDeviceOwnerId);
+
+        //Act
+        PortEntity updatedPort = this.redirApi.updatePort(portEntity);
+
+        //Assert
+        assertNotNull(updatedPort);
+        assertEquals(newDeviceOwnerId, updatedPort.getDeviceOwnerId());
+        assertEquals(newParentId, updatedPort.getParentId());
+        assertEquals(macAdresses, updatedPort.getMacAddresses());
+        assertEquals(ipAdresses, updatedPort.getPortIPs());
+    }
+
+    @Test
+    public void testApiGetInspectionPortsIds() throws Exception {
+        //Arrange
+        this.redirApi = new SampleSdnRedirectionApi(this.txControl, this.em);
+
+        InspectionPortElement inspectionPortElement = new InspectionPortEntity(null, ingress, egress);
+        inspectionPortElement = (InspectionPortElement) this.redirApi.registerInspectionPort(inspectionPortElement);
+
+        //Act
+        final String inspectionPortElementId = inspectionPortElement.getElementId();
+
+        //Assert
+        assertEquals(1, this.redirApi.getInspectionPortsIds().size());
+        assertTrue(this.redirApi.getInspectionPortsIds().contains(inspectionPortElementId));
+    }
+
+    @Test
+    public void testApiGetInspectionHooksIds() throws Exception {
+        //Arrange
+        this.redirApi = new SampleSdnRedirectionApi(this.txControl, this.em);
+
+        InspectionPortEntity inspectionPortElement = new InspectionPortEntity(null, ingress, egress);
+
+        Element registeredElement = this.redirApi.registerInspectionPort(inspectionPortElement);
+
+        assertNotNull(registeredElement);
+        assertNotNull(registeredElement.getElementId());
+
+        //Act
+        final String hookId = this.redirApi.installInspectionHook(inspected, inspectionPortElement, 0L, VLAN, 0L, NA);
+
+        //Assert
+        assertNotNull(hookId);
+        assertTrue(this.redirApi.getInspectionHooksIds().contains(hookId));
+        assertEquals(1, this.redirApi.getInspectionHooksIds().size());
+    }
+
+    @Test
+    public void testApiUpdateInspectionPort() throws Exception {
+        //Arrange
+        final ArrayList<String> newIngressMacAdresses = new ArrayList<String>(
```

Comment:
Class variables?

Reply:
removed final and retained as a local variables in the method.